src/libstd/fs.rs

```diff
@@ -1069,13 +1069,23 @@ pub fn remove_dir_all<P: AsRef<Path>>(path: P) -> io::Result<()> {
 }
 
 fn _remove_dir_all(path: &Path) -> io::Result<()> {
+    let filetype = try!(symlink_metadata(path)).file_type();
+    if filetype.is_symlink() {
+        remove_file(path)
```

Comment:
> I think that this may want to also use is_dir to choose whether to recurse or not (like below). Junctions on Windows (e.g. hard links to directories) don't report is_dir, but to behave the same as unix we should delete their contents.

Isn't it a contradiction? If they don't report `is_dir` how should we know to recurse?

The question probably is: `is_dir() && is_symlink()` (impossible on unix) should recurse or `remove_dir()` ?